hazelcast/src/test/java/com/hazelcast/test/starter/constructor/ConfigConstructor.java

```diff
@@ -16,29 +16,16 @@
 
 package com.hazelcast.test.starter.constructor;
 
-import com.hazelcast.nio.ClassLoaderUtil;
+import com.hazelcast.test.starter.HazelcastStarterConstructor;
 
-import java.lang.reflect.Constructor;
-import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Properties;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
-
-import static com.hazelcast.test.starter.HazelcastProxyFactory.ProxyPolicy.RETURN_SAME;
-import static com.hazelcast.test.starter.HazelcastProxyFactory.shouldProxy;
-import static com.hazelcast.test.starter.HazelcastStarterUtils.debug;
-import static com.hazelcast.test.starter.ReflectionUtils.getFieldValueReflectively;
 
 /**
  * Clones the configuration from {@code mainConfig} to a new configuration object loaded in the
  * target {@code classloader}. The returned configuration has its classloader set to the target classloader.
  */
-public class ConfigConstructor extends AbstractStarterObjectConstructor {
+@HazelcastStarterConstructor(classNames = {"com.hazelcast.config.Config", "com.hazelcast.client.config.ClientConfig"})
```

Comment:
Example usage with multiple classes (the `NO_PROXY` is the default, so you just have to define the class names).